Lib/asyncio/tasks.py

```diff
@@ -107,11 +109,17 @@ def __del__(self):
             if self._source_traceback:
                 context['source_traceback'] = self._source_traceback
             self._loop.call_exception_handler(context)
-        futures.Future.__del__(self)
+        super().__del__()
 
     def _repr_info(self):
         return base_tasks._task_repr_info(self)
 
+    def set_result(self, result):
+        raise NotImplementedError
+
+    def set_exception(self, exception):
+        raise NotImplementedError
```

Comment:
It's backward incompatible change.
I know people use `task.set_exception()` in production code, heard about it on conference couple months ago.
I propose raising a deprecation warning in 3.7 and explicit exception (I thought about`RuntimeError` but `NotImplementedeError` is also good maybe) in 3.8.

Reply:
But it's impossible to use it, i've shown that in the issue. That code is almost certainly broken. Maybe it's OK to force people to fix nonsense code?

Reply:
Let's pause for day or two: I'm going to wrote email to that guy for clarification.

Reply:
Maybe @kozzztik can comment his use case.

Reply:
Hi! Thats correct, we used it on prod for monitoring purposes. However as said above it was broken, thats what I said on conference. When you try to inherit from Task on Py>3.6 (on 3.5 it is ok) that is not obvious, but you inherit CTask which not uses this method.
However, I guess why method raises "Not implemented" and called with magic super just bellow declaration. It looks strange. I see two solutions that looks better:
1. Just remove it (may be also make C implementation be similar to python variant and respect method overriding)
2. Add comment to "raise" statement why it looks so. Sure, someone will spend hours to understand what that mean.

Reply:
@kozzztik Thanks for the explanation!

> When you try to inherit from Task on Py>3.6 (on 3.5 it is ok) that is not obvious, but you inherit CTask which not uses this method.

@asvetlov As I expected.  It's not possible to use either method since 3.6.

> Just remove it (may be also make C implementation be similar to python variant and respect method overriding)

Can't do that, Task inherits from Future.  It's possible to remove them from C Task, but not from Python version.

> Add comment to "raise" statement why it looks so. Sure, someone will spend hours to understand what that mean.

OK.

I'll change `NotImplementedError` to some other exception.